### PR TITLE
fix hdr issue

### PR DESCRIPTION
**Pull request by Dylan-debug**
fix hdr local variable pointer access issue

local variable will be freed after return, so need the
caller to convey it

Signed-off-by: caij <jianxing.cai@intel.com>

### CODE REVIEW

File: videoprocess/vpphdr_tm.cpp [hunk 1]
```diff
@@ -296,7 +295,8 @@ video_frame_process(VASurfaceID in_surface_id,
     }
 
     hdrtm_filter_init(&filter_param_buf_id, g_tm_type);
-    hdrtm_metadata_init(out_metadata, g_tm_type);
+    VAHdrMetaDataHDR10 *out_hdr10_metadata = (VAHdrMetaDataHDR10*)malloc(sizeof(VAHdrMetaDataHDR10));
```
Comment (FurongZhang) [inline videoprocess/vpphdr_tm.cpp:298]: Did not free it?

Comment (LhGu) [reply videoprocess/vpphdr_tm.cpp:298]: Maybe we can just use stack var here. VAHdrMetaDataHDR10 out_hdr10_metadata = {} and pass its address to hdrtm_metadata_init function

Comment (LhGu) [reply videoprocess/vpphdr_tm.cpp:298]: Or just using  static var  out_hdr10_metadata may also work for this cpp main file.

Comment (Dylan-debug) [reply videoprocess/vpphdr_tm.cpp:298]: > Maybe we can just use stack var here. VAHdrMetaDataHDR10 out_hdr10_metadata = {} and pass its address to hdrtm_metadata_init function

agree

File: videoprocess/vpphdr_tm.cpp [hunk 2]
```diff
@@ -217,48 +217,47 @@ hdrtm_filter_init(VABufferID *filter_param_buf_id, uint32_t tm_type)
...
 {
     VAStatus va_status = VA_STATUS_SUCCESS;
 
-    VAHdrMetaDataHDR10 out_hdr10_metadata = {}; 
 
-    out_hdr10_metadata.max_display_mastering_luminance = g_out_max_display_luminance;
-    out_hdr10_metadata.min_display_mastering_luminance = g_out_min_display_luminance;
-    out_hdr10_metadata.max_content_light_level         = g_out_max_content_luminance;
-    out_hdr10_metadata.max_pic_average_light_level     = g_out_pic_average_luminance;
+    out_hdr10_metadata->max_display_mastering_luminance = g_out_max_display_luminance;
```
Comment (FurongZhang) [inline videoprocess/vpphdr_tm.cpp:232]: Compiler issue.

Comment (Dylan-debug) [reply videoprocess/vpphdr_tm.cpp:232]: done

**Review by FurongZhang — APPROVED** [review]

**Review by FurongZhang — APPROVED** [review]

**Comment by FurongZhang** [comment]
Looks good to me.

File: videoprocess/vpphdr_tm.cpp [hunk 3]
```diff
@@ -557,6 +557,11 @@ bool read_frame_to_surface(FILE *fp, VASurfaceID surface_id)
         u_size = (va_image.width / 2 * bytes_per_pixel) * (va_image.height >> 1);
 
         src_buffer = (unsigned char*)malloc(frame_size);
+        if (src_buffer)
+        {
+            printf("malloc buffer failed \n");
+            return false;
+        }
```
Comment (LhGu) [inline videoprocess/vpphdr_tm.cpp:564]: If malloc failed, maybe we also need to do some clean job(vaUnmapBuffer and vaDestroyImage)

Comment (Dylan-debug) [reply videoprocess/vpphdr_tm.cpp:564]: ooh! yes, Thanks for reminding.

**Review by LhGu — APPROVED** [review]

**Pull request merged**